### PR TITLE
Add accessibility needs to locker_application

### LINKED ISSUE #125: Allow users to request specific accessibility needs
### User story
As a student with both visual impairment and mobility accessibility needs, I'd like to communicate my specific accessibility needs when requesting a locker.

### Concrete example
It takes different accommodations in order to make a locker accessible to someone. 

We can offer a checklist of known accommodations, including being near and elevator, keyed entry rather than combination, as well as adding a write-in space for accommodations we may be able to provide for but may not be able to anticipate ahead of time, such as a locker in a quieter space for people with sensory sensitivities. 

### Implementation notes, if any
This could be a multi-select list with an additional "other" write-in field.

### PR DESCRIPTION
**Pull request by maxkadel**
- Currently multi-select but does not allow user to type in answer

Co-authored-by: Ryan Laddusaw <rladdusaw@users.noreply.github.com>

Closes #125

screenshot of application form with list of accessibility accommodations

screenshot of application show page with list of requested accessibility accommodations

### CODE REVIEW

**Review by sandbergja — CHANGES_REQUESTED** [review]
Looks good, @maxkadel .  I have one concern, though.  Any existing applications where the "accessible" checkbox was checked now no longer display on the locker assignment page.

For example: https://lockers-and-study-spaces-staging.princeton.edu/locker_applications/835/assign -- the user requested an accessible locker before this branch was deployed, but it doesn't display on this screen, so the staff member wouldn't know about their request.

**Comment by maxkadel** [comment]
> Looks good, @maxkadel . I have one concern, though. Any existing applications where the "accessible" checkbox was checked now no longer display on the locker assignment page.
> 
> For example: https://lockers-and-study-spaces-staging.princeton.edu/locker_applications/835/assign -- the user requested an accessible locker before this branch was deployed, but it doesn't display on this screen, so the staff member wouldn't know about their request.

Thanks @sandbergja , that's a good point. I can see two ways of addressing that:

1. Add a migration that adds unspecified "Accessible: true" to some value in the new `accessibility_needs` field - e.g. `['Unspecified accessibility need]` or something like that
2. Display both the old Accessible boolean and the new accessibility_needs array on the show view

**Comment by sandbergja** [comment]
Thanks, @maxkadel !

>     1. Add a migration that adds unspecified "Accessible: true" to some value in the new `accessibility_needs` field - e.g. `['Unspecified accessibility need]` or something like that

I like option 1 better — it seems like a less complex ui, and make it easier if we wanted to remove the boolean field eventually.

File: app/models/locker_application.rb [hunk 1]
```diff
@@ -69,4 +69,12 @@ def self.search(uid:, archived:)
 
     where(complete: true)
   end
+
+  def self.migrate_accessible_field
+    LockerApplication.where(accessible: true).find_each do |application|
```
Comment (sandbergja) [inline app/models/locker_application.rb:74]: I believe the accessibility info is used only when assigning lockers.  So, if we don't want to touch as many rows in the migration, we could just select applications that don't yet have an assignment.

**Review by sandbergja — COMMENTED** [review]
Just one thought on the migration before running it on staging.  Not a blocker.

**Review by sandbergja — CHANGES_REQUESTED** [review]
Thanks, @maxkadel !  Two smaller issues I found in the UI today:

- Extra bullet point in the list of accessibility needs at https://lockers-and-study-spaces-staging.princeton.edu/locker_applications/860/assign
- When you attempt to edit an existing application with accessibility needs specified (e.g. https://lockers-and-study-spaces-staging.princeton.edu/locker_applications/860/edit), the existing accessibility needs do not display

**Review by sandbergja — APPROVED** [review]
Great feature, nice implementation.  Thanks, @maxkadel !  🦄🦄🦄

**Pull request merged**